### PR TITLE
Changes to remove TODOs

### PR DESCRIPTION
**Pull request by anija-anil**
To Remove TODOs for https://github.com/OpenLiberty/open-liberty/issues/27501

for #27501
for #23952 
for #23944
for #23566

### CODE REVIEW

**Comment by anija-anil** [comment]
#build

**Comment by anija-anil** [comment]
#build

**Comment by anija-anil** [comment]
#libby

File: dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import java.security.PrivilegedAction;
 import java.util.Map;
 
+import javax.management.Query;
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java]: This is causing a compile failure... the class `Query` cannot be imported twice from 2 different packages.  The existing import on line 23 is correct, please remove this new one.

File: dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java [hunk 2]
```diff
@@ -91,7 +92,7 @@ public void testVersioning001(TestExecutionContext testExecCtx, TestExecutionRes
          * https://github.com/OpenLiberty/open-liberty/issues/23952
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java]: Please remove the TODO comment for 23952, as that is the TODO that is being fixed here.

File: dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java [hunk 3]
```diff
@@ -91,7 +92,7 @@ public void testVersioning001(TestExecutionContext testExecCtx, TestExecutionRes
...
          */
         if (isDB2ZOS && JPAPersistenceProvider.ECLIPSELINK.equals(provider)
-            && (isUsingJPA21Feature() || isUsingJPA31Feature())) {
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java:94]: This same TODO exists 2 more times in this file... all 3 TODOs need to be removed.

File: dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java [hunk 4]
```diff
@@ -6579,11 +6579,6 @@ public void testROUND2_PartialBind_BindLiteral(TestExecutionContext testExecCtx,
         boolean isDerby = DatabaseVendor.checkDBProductName(dbProductName, dbProductVersion, DatabaseVendor.DERBY);
         boolean isPostgres = DatabaseVendor.checkDBProductName(dbProductName, dbProductVersion, DatabaseVendor.POSTGRES);
 
-        // TODO: Disable until https://github.com/OpenLiberty/open-liberty/issues/23566 is delivered
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:6582]: This same TODO occurs 5 more times in this file.... all 6 need to be remove.

**Review by tkburroughs — CHANGES_REQUESTED** [review]

**Comment by anija-anil** [comment]
#build

File: dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java (hunk 1 above)
Comment (tkburroughs) [reply dev/com.ibm.ws.jpa.tests.spec10.entity_fat.common/test-applications/entity/src/com/ibm/ws/jpa/fvt/entity/testlogic/VersioningTestLogic.java]: This is still a problem... the test requires `javax.persistence.Query`.... replacing it with `javax.management.Query` will not work, they are totally different things.  Need to remove `javax.management.Query` and restore the original `javax.persistence.Query`

File: dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java [hunk 5]
```diff
@@ -17,6 +17,8 @@
 import java.util.List;
 import java.util.Map;
 
+import javax.management.Query;
```
Comment (tkburroughs) [inline dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java]: The code is failing to compile... the test requires `javax.persistence.Query`.... replacing it with `javax.management.Query` will not work, they are totally different things.  Need to remove `javax.management.Query` and restore the original `javax.persistence.Query`

**Review by tkburroughs — CHANGES_REQUESTED** [review]
Still need to fix the imports for `javax.persistence.Query`. Not sure why your editor is trying to replace it, but `javax.management.Query` is not compatible with `javax.persistence.Query`... they are totally different things.

**Comment by tkburroughs** [comment]
From the build; look in the `gradle_comipleTest.log` and there are compile failures like this:

```
> Task :com.ibm.ws.jpa.tests.eclipselink.query_jpa_2.1_fat:globalize NO-SOURCE
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:78: error: incompatible types: jakarta.persistence.Query cannot be converted to javax.management.Query
            Query query = em.createQuery("SELECT CEILING(?1) FROM JPA31Entity s WHERE s.intVal1 = ?2");
                                        ^
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:79: error: cannot find symbol
            query.setParameter(1, 1.1);
                 ^
  symbol:   method setParameter(int,double)
  location: variable query of type Query
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:80: error: cannot find symbol
            query.setParameter(2, 3);
                 ^
  symbol:   method setParameter(int,int)
  location: variable query of type Query
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:81: error: cannot find symbol
            query.getResultList();
                 ^
  symbol:   method getResultList()
  location: variable query of type Query
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:90: error: incompatible types: jakarta.persistence.Query cannot be converted to javax.management.Query
            query = em.createQuery("SELECT CEILING(1.1) FROM JPA31Entity s WHERE s.intVal1 = 3");
                                  ^
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:91: error: cannot find symbol
            query.getResultList();
                 ^
  symbol:   method getResultList()
  location: variable query of type Query
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:110: error: incompatible types: no instance(s) of type variable(s) T exist so that TypedQuery<T> conforms to Query
            query = em.createQuery(cquery);
                                  ^
  where T is a type-variable:
    T extends Object declared in method <T>createQuery(CriteriaQuery<T>)
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:111: error: cannot find symbol
            query.setParameter(floatParam1, 1.1f);
                 ^
  symbol:   method setParameter(ParameterExpression<Float>,float)
  location: variable query of type Query
/home/jazz_build/Build/jbe/build/open-liberty/dev/com.ibm.ws.jpa.tests.eclipselink.query_31_fat.common/test-applications/jpa31/src/com/ibm/ws/jpa/jpa31/testlogic/TestMathLogic.java:112: error: cannot find symbol
            query.setParameter(intParam1, 3);
                 ^
  symbol:   method setParameter(ParameterExpression<Integer>,int)
  location: variable query of type Query
```

The change to use `javax.management.Query` does not work; it is not compatible with `javax.persistence.Query`; the correct import needs to be restored.

**Comment by anija-anil** [comment]
#build

**Comment by tkburroughs** [comment]
The "import" statements are still incorrect.  The builds will continue to fail as long as the "[Files changed](https://github.com/OpenLiberty/open-liberty/pull/28066/files)" shows changes to the imports.

**Comment by anija-anil** [comment]
#build

**Comment by tkburroughs** [comment]
#libby

**Comment by anija-anil** [comment]
#build

**Review by tkburroughs — APPROVED** [review]

**Comment by tkburroughs** [comment]
The build had one test failure that is a known intermittent unrelated failure; merged.

**Pull request merged**